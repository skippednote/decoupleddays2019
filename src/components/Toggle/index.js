import React, { useState } from "react";

function Toggle({ children, on, off, ...rest }) {
  const [state, setState] = useState(false);
  return (
    <div>
      <button
        data-testid="show-filters"
        className="button"
        onClick={() => setState(!state)}
      >
        {state ? on : off}
      </button>
      {state ? children : null}
    </div>
  );
}

export default Toggle;
