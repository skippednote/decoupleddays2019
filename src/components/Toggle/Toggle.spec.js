import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Toggle from ".";

describe("toggle", () => {
  it("should have a toggle button", () => {
    const { getByText, container } = render(<Toggle on="Hide Filter" off="Show Filter">Hello</Toggle>);
    getByText("Show Filter");
    expect(container).toMatchSnapshot();
  });

  it("should toggle text on button click", () => {
    const { getByText, container } = render(<Toggle on="Hide Filter" off="Show Filter">Hello</Toggle>);
    fireEvent.click(getByText("Show Filter"));
    expect(getByText("Hide Filter")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should toggle children on click", () => {
    const { queryByText, getByText } = render(<Toggle on="Hide Filter" off="Show Filter">Hello</Toggle>);
    expect(queryByText("Hello")).toBeNull();
    fireEvent.click(getByText("Show Filter"));
    expect(queryByText("Hello")).toBeInTheDocument();
  });
});
