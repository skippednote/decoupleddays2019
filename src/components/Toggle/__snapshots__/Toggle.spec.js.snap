// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle should have a toggle button 1`] = `
<div>
  <div>
    <button
      class="button"
      data-testid="show-filters"
    >
      Show Filter
    </button>
  </div>
</div>
`;

exports[`toggle should toggle text on button click 1`] = `
<div>
  <div>
    <button
      class="button"
      data-testid="show-filters"
    >
      Hide Filter
    </button>
    Hello
  </div>
</div>
`;
